src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -364,11 +376,219 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
             }
         }
 
+        private ImmutableArray<BoundExpression> VisitArguments(
+            SyntaxNode syntax,
+            ImmutableArray<BoundExpression> arguments,
+            Symbol methodOrIndexer,
+            ImmutableArray<int> argsToParamsOpt,
+            ref ImmutableArray<RefKind> argumentRefKindsOpt,
+            [NotNullIfNotNull("visitedReceiver")] ref BoundExpression? rewrittenReceiver,
+            out ArrayBuilder<LocalSymbol>? temps,
+            out BitVector positionsAssignedToTemp,
+            out bool receiverIsArgumentSideEffectSequence)
+        {
+            Debug.Assert(argumentRefKindsOpt.IsDefault || argumentRefKindsOpt.Length == arguments.Length);
+            var requiresInstanceReceiver = methodOrIndexer.RequiresInstanceReceiver() && methodOrIndexer is not MethodSymbol { MethodKind: MethodKind.Constructor };
+            Debug.Assert(!requiresInstanceReceiver || rewrittenReceiver != null || _inExpressionLambda);
+            temps = null;
+            positionsAssignedToTemp = BitVector.Null;
+            receiverIsArgumentSideEffectSequence = false;
+
+            if (arguments.IsEmpty)
+            {
+                return arguments;
+            }
+
+            var visitedArgumentsBuilder = ArrayBuilder<BoundExpression>.GetInstance(arguments.Length);
+            ArrayBuilder<BoundExpression>? sideEffectsBuilder = null;
```

Comment:
>sideEffectsBuilder [](http://example.com/codeflow?start=43&length=18)

I don't think we really need to track any sideffects across all arguments. Each argument  and receiver can be captured individually and can keep the store in a sequence in place.